### DISCUSSION TITLE
How to listen to a Provider in the initState method

### DISCUSSION THREAD

**Discussion by json1994** [rrousselGit/riverpod#3511] [Q&A]
I have defined a StateProvider to listen for changes in the user token (user login or logout), and when the token changes, I need to create different gRPC services. How should I listen for this change? 
`final userDataProvider =
    StateNotifierProvider<UserNotifier, UserEntity?>((ref) {
  return UserNotifier(app.userEntity);
});

final signUpProvider = StateProvider.autoDispose<String?>((ref) {
  final ret = ref.watch(userDataProvider);
  ref.onDispose(() {
    logger?.i("signUpProvider has ben dispose");
  });
  return ret?.token;
});
<img width="940" alt="iShot_2024-04-28_22 40 54" src="https://github.com/rrousselGit/riverpod/assets/34496479/6d6202a3-4547-455a-af48-ecf88e1b32da">
<img width="682" alt="iShot_2024-04-28_22 41 59" src="https://github.com/rrousselGit/riverpod/assets/34496479/f302cd97-6ec0-4e9b-b395-96bc001473d4">
`

**Comment by rrousselGit** [comment]
Use ref.listenManual(provider, ...)
